File: 07_binary_search/binary_search.py
def binary_search(x, key):
    low = 0
    high = len(x) - 1

    while low <= high:
        mid = (high + low) // 2
        # print(mid)
        if key == x[mid]: 
            return mid
        else:
            if key > x[mid]:
                low = mid + 1
            else:
                high = mid - 1
    return -1;
test_array =[1,5,6,7,10,15]
print(binary_search(test_array, 7))

File: 07_binary_search/binary_search.js
let x = [1, 2, 5, 7, 10, 15];

function binarySearch(arr, key) {
  let low = 0;
  let high = arr.length - 1;

  while (low <= high) {
    let mid = Math.floor((low + high) / 2);

    if (arr[mid] === key) {
      return mid;
    }
    if (key > arr[mid]) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }
  }
  return -1;
}
console.log(binarySearch(x, 7));
